File: CPP/Creational/Builder/Example/Product.cpp
//
// Created by ray on 4/16/25.
//
#include "Product.h"
#include <iostream>

void Product::addPart(const std::string& part) {
    parts.push_back(part);
}

void Product::show() const {
    std::cout << "Product parts: ";
    for (const auto& part : parts) {
        std::cout << part << " ";
    }
    std::cout << std::endl;
}

File: TheRayCode/CSharp/Creational/Builder/Example/Example/Product.cs
namespace Example;

public class Product
{
    private List<string> _parts = new List<string>(); // List to store product parts

    // Adds a part to the product
    public void Add(string part)
    {
        _parts.Add(part);
    }

    // Display all parts of the product
    public void Show()
    {
        Console.WriteLine("Product Parts:");
        foreach (string part in _parts)
        {
            Console.WriteLine("- " + part);
        }
    }

}